### DISCUSSION TITLE
HALLUCINATION

### DISCUSSION THREAD

**Discussion by mukesh1811** [vanna-ai/vanna#556] [General]
we are facing a critical hallucination problem where the generated sql is having hallucinated tables and columns. there tables and columns are not available in my schema. how do I tackle that? I am using gpt-3.5-turbo-16k

**Comment by aslanok** [comment]
Hi, I am facing with same situation. Can you solve the problem ? I am using local LLM gemma2:27b.

**Reply by zainhoda** [reply]
@aslanok For your case read this thread:
https://x.com/zain_hoda/status/1836432154558902740

**Reply by CodingNowPls** [reply]
> Hi, I am facing with same situation. Can you solve the problem ? I am using local LLM gemma2:27b.

 bro，you can look at my example below, it seems to solve this problem,you need to tell big model don't modfiy you provide the table name and column name ,otherwise ,the big model may be modify them

**Comment by CodingNowPls** [comment]


**Reply by wangquan3294** [reply]
你好，大佬，我用这个ollama搭载的千问模型，发现这个vanna很不好用，即使是增加了qa对，也没办法识别，你有解决办法吗

**Reply by CodingNowPls** [reply]
> 你好，大佬，我用这个ollama搭载的千问模型，发现这个vanna很不好用，即使是增加了qa对，也没办法识别，你有解决办法吗

我用着还行，你得使用向量rag搜索一条最近的，然后让大模型回答，他这个对中文不友好，你得使用中文向量，我在上面代码应该有写

**Reply by wangquan3294** [reply]
可以加个微信探讨下吗，wq03077

**Reply by wangquan3294** [reply]
![微信截图_20241011171437](https://github.com/user-attachments/assets/2fa76ddd-58ae-4e89-ad26-8b0de8b9f2b2)



![微信截图_20241011171449](https://github.com/user-attachments/assets/1b8f7c14-ced2-4aa9-9b97-f8de4f8980c8)

**Reply by wangquan3294** [reply]
没有作用

**Reply by CodingNowPls** [reply]
> 没有作用

好的，我加你了

**Comment by CodingNowPls** [comment]
this is my project repo : https://github.com/CodingNowPls/vanna-flask-rest-cn
